### DISCUSSION TITLE
[Art competition] Context-sensitive analysis

### DISCUSSION THREAD

**Discussion by KunJeong** [prosyslab-classroom/cs524-program-analysis#279] [Art competition]
![DALL·E 2022-11-30 15 58 03](https://user-images.githubusercontent.com/10477004/204728843-151bb079-4d41-4ce8-962e-f6a66831d0a1.png)

## Short Description
For me, [Context-sensitive analysis] means [a landscape full of different circular portals to another universe, some of them inside other portals, digital art]

## Long Description
Function calls send values to a completely different location in the program, which is similar to jumping through portals. By using context-sensitive analysis, we can prove more assertions about a program by maintaining the call stack of functions to a certain depth. This is similar to each portal maintaining its form instead of disappearing after use (=> a land full of portals), and being able to see through multiple stacked portals.

**Comment by KihongHeo** [comment]
Then, the k-th portal is the event horizon?
Why don't you make a song like [this](https://www.youtube.com/watch?v=73V3xrfiYMo)?

**Comment by KunJeong** [comment]
Yes, with k-sensitivity that would be the case
